lib/corefoundation/exceptions.rb

```diff
@@ -0,0 +1,16 @@
+module CF
+  module Exceptions
+    # Raised when the preference value couldn't be found for a domain.
+    class PreferenceDoesNotExist < RuntimeError
+      def initialize(key, domain, hostname)
+        super("Returned NULL value for \"#{key}\" in \"#{domain}\", hostname: #{hostname}")
+      end
+    end
+    # Raised when the preference value failed to write.
+    class PreferenceSyncFailed < RuntimeError
```

Comment:
All of the above/exception module's comments apply here too.
Also, error names have to follow ruby convention, i.e., `Error` in suffix, so something like `PreferenceSyncError`

Reply:
This is more of a convention that is seen in many open source ruby projects. But I guess we already have many exceptions in chef that don't follow this, so I am happy to let this pass.